### PR TITLE
Store binlog position in database

### PR DESCRIPTION
**Pull request by osheroff**
@vanchi-zendesk this is a first draft of storing the current binlog position in the database.

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/Maxwell.java [hunk 1]
```diff
@@ -35,6 +34,8 @@ private void initFirstRun() throws SQLException, IOException {
 	private void run(String[] args) throws Exception {
 		this.config = MaxwellConfig.buildConfig("config.properties", args);
 
+		SchemaStore.createMaxwellSchema(this.config.getMasterConnection());
+
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/Maxwell.java]: Don't understand this. This needs to happen only exactly once right? So it was at it's right place in first run?

Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/Maxwell.java]: it needs to happen before `getInitialPosition()...`

Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/Maxwell.java]: and it's a misnomer, it should be called `createIfNotExsistsMaxwellSchema`...

File: src/main/java/com/zendesk/maxwell/MaxwellParser.java [hunk 2]
```diff
@@ -92,17 +92,23 @@ public void run() throws Exception {
...
-			this.config.setInitialPosition(p);
+				BinlogPosition p = eventBinlogPosition(event);
+				this.config.setInitialPosition(p);
+			}
 		}
 	}
 
+	private boolean skipEvent(MaxwellAbstractRowsEvent event) {
+		return event.getTable().getDatabase().getName().equals("maxwell");
+	}
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: If you think about this a bit, this is fine. 

Migration -> maxwell writes -> kafka push. (so if we follow maxwell, we get the schema changes as snapshots)

But I think it's best to not complicate at the beginning :)

File: src/main/java/com/zendesk/maxwell/schema/SchemaPosition.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+		return new BinlogPosition(rs.getLong("binlog_position"), rs.getString("binlog_file"));
+	}
+
+	public static void set(Connection c, Long serverID, BinlogPosition p) throws SQLException {
+		String sql = "INSERT INTO `maxwell`.`positions` set "
+				+ "server_id = ?, "
+				+ "binlog_file = ?, "
+				+ "binlog_position = ? "
+				+ "ON DUPLICATE KEY UPDATE binlog_file=?, binlog_position=?";
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/schema/SchemaPosition.java]: could we use REPLACE INTO and save lines?

File: src/main/java/com/zendesk/maxwell/Maxwell.java (hunk 1 above)
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/Maxwell.java]: right. I suspected.

Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/Maxwell.java]: `EnsureMaxwellSchema`

**Comment by vanchi-zendesk** [comment]
:+1:

File: src/main/java/com/zendesk/maxwell/MaxwellParser.java (hunk 2 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: not with the introduction of the maxwell.position table, which would keep the system in an infinite loop

File: src/main/java/com/zendesk/maxwell/schema/SchemaPosition.java (hunk 3 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/schema/SchemaPosition.java]: on duplicate key is way way faster:

http://stackoverflow.com/questions/9168928/what-are-practical-differences-between-replace-and-insert-on-duplicate-ke

**Pull request merged**